src/actions/bitcoin-kit.ts

```diff
@@ -0,0 +1,84 @@
+import { type Chain, type Client, type Hash, type Transport } from "viem"
+import { readContract } from 'viem/actions'
+
+import {
+  bitcoinKitTxsAbi,
+  bitcoinKitTxAddresses,
+} from "../contracts/bitcoin-kit-txs.js"
+
+export function getBitcoinAddressBalance<
+  TTransport extends Transport = Transport,
+  TChain extends Chain | undefined = Chain | undefined
+>(client: Client<TTransport, TChain>, parameters: { btcAddress: string }) {
+  const { btcAddress } = parameters
+  return readContract(client, {
+    address: bitcoinKitTxAddresses[client.chain!.id],
+    abi: bitcoinKitTxsAbi,
+    functionName: "getBitcoinAddressBalance",
+    args: [btcAddress],
+  })
+}
+
+export function getHeaderN<
+  TTransport extends Transport = Transport,
+  TChain extends Chain | undefined = Chain | undefined
+>(client: Client<TTransport, TChain>, parameters: { height: number }) {
```

Comment:
The `height` in the contract is an `uint32`. Should we use `bigint` here too?

```suggestion
>(client: Client<TTransport, TChain>, parameters: { height: bigint }) {
```

Reply:
`viem` (through the readContract inference) infers that `height` must be a number. It does so from the ABI. If I set `height` as a `bigint`, it fails to compile. See 

![image](https://github.com/user-attachments/assets/54740b13-0d33-4eee-960b-925ec4b4e077)

![image](https://github.com/user-attachments/assets/507cddc4-f9ac-47e0-b155-500ba384a1e1)


It seems it maps `uint256` to bigint, but `uint32` to `number`

Reply:
I am not sure if this line is exactly used when calling `readContract`, but [here](https://github.com/wevm/viem/blob/bc419b7d2b7c2f8b5ea44a3859bb59cb8b5bbf01/src/utils/abi/decodeAbiParameters.ts#L270) maps ints to BigInt if the associated size is bigger than 48

Reply:
You are right. The ABI says `uint32` so a `number` is ok.